controller/handler_ctrl/connect.go

```diff
@@ -98,16 +61,51 @@ func (self *ConnectHandler) HandleConnection(hello *channel2.Hello, certificates
 		KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
 	}
 
+	var validFingerPrints []string
 	var errorList errorz.MultipleErrors
 
-	for _, cert := range certificates {
+	for i, cert := range certificates {
 		if _, err := cert.Verify(opts); err == nil {
-			return nil
+			fingerprint := fmt.Sprintf("%x", sha1.Sum(cert.Raw))
+			validFingerPrints = append(validFingerPrints, fingerprint)
+			log.Debugf("%d): peer certificate fingerprint [%s]", i, fingerprint)
+			log.Debugf("%d): peer common name [%s]", i, cert.Subject.CommonName)
 		} else {
 			errorList = append(errorList, err)
 		}
 	}
 
-	//goland:noinspection GoNilness
-	return errorList.ToError()
+	if len(errorList) > 0 {
+		return errorList.ToError()
```

Comment:
check for valid certs/fingerprints instead?
this will exit and ignore valid cert(s) if there is an invalid one.

Reply:
Fixed. Check is now if there are valid fingerprints